bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -12932,8 +12954,11 @@ public void rollback() throws TskCoreException {
 		 *
 		 */
 		void close() {
-			this.connection.close();
-			sleuthkitCase.releaseSingleUserCaseWriteLock();
+			synchronized(threadsWithOpenTransactionLock) {
+				this.connection.close();
+				sleuthkitCase.releaseSingleUserCaseWriteLock();
+				threadsWithOpenTransaction.remove(Thread.currentThread().getId());
+			}
```

Comment:
Can the scope of this synchronized block be reduced to just the call into the Map protected by the lock Object?